src/qtcomponents.h

```diff
@@ -0,0 +1,29 @@
+#ifndef QTCOMPONENTS_H
+#define QTCOMPONENTS_H
+
+#include <QKeyEvent>
+#include <QTreeView>
+
+typedef void ( * TreeViewKeyPressedEvent)(void * handle, QKeyEvent * event);
+
+/**
+ * A QTreeView that allow to define listener of keyEvent received
+ */
+class QTreeViewWithKeyEvents : public QTreeView {
+    Q_OBJECT
+
+    public:
+        explicit QTreeViewWithKeyEvents(QWidget *parent = NULL);
+
+    public slots:
+        void onKeyPressed(void * handleUsed, TreeViewKeyPressedEvent callback);
+
+    protected:
+        void keyPressEvent(QKeyEvent * event) override;
+
+    private:
+        TreeViewKeyPressedEvent mCallback;
+        void * mHandle;
+};
```

Comment:
How about we just use the Qt event system instead of building our own, that way we can avoid some casts and manually storing listeners.

```suggestion
/**
 * A QTreeView that allow to define listener of keyEvent received.
 */
class QTreeViewWithKeyEvents : public QTreeView {
Q_OBJECT

public:
    explicit QTreeViewWithKeyEvents(QWidget* parent = nullptr);

signals:
    void onKeyPressed(QKeyEvent* event);

protected:
    void keyPressEvent(QKeyEvent* event) override;
};
```

with

```c++
void QTreeViewWithKeyEvents::keyPressEvent( QKeyEvent *event )
{
    emit onKeyPressed(event);
    QTreeView::keyPressEvent(event);
}
```

and

```c++
connect(treeAccounts, &QTreeViewWithKeyEvents::onKeyPressed, this, &DialogSettings::onKeyPressedOnTreeAccount);
```

Reply:
Looks fine with me. I mean, those are rare operations, hardly worthy optimizing, but as you wish :)